src/api/python/mkenums.py

```diff
@@ -0,0 +1,145 @@
+#
+# Generate pyboolector_enums.pxd from btortypes.h
+#
+# Usage: mkenums.py path/to/btortypes.h path/to/outputfile
+#
+
+import sys
+import os
+from collections import OrderedDict
+
+
+def extract_enums(btortypes):
+    """
+    Given the file "btortypes.h", constructs a dictionary representing the set
+    of all Boolector enums and corresponding values
+    """
+
+    # We want to get only non-empty, stripped lines from our input file
+    lines = [
+        line
+        for line in [line.strip() for line in open(btortypes, "r").readlines()]
+        if line
+    ]
+
+    # Manually create an iterator as we want to be able to move this forwards ourselves
+    line_iter = iter(lines)
+
+    # Dictionary of all our enums and their values
+    btor_enums = OrderedDict()
+
+    # For each line ...
+    for line in line_iter:
+
+        # Do we see the _start_ of an enum?
+        if line.startswith("enum Btor"):
+
+            # Obtain the enum name by splitting the line and taking the second
+            # part
+            enum_name = line.split(" ")[1]
+
+            # Get the next line
+            line = next(line_iter)
+
+            # We expect this to be an opening curly, given Boolector's style
+            if line == "{":
```

Comment:
Is this check required? I'm a little bit concerned relying on the code style to collect the enums - with a different style enums would be skipped.

Reply:
How about changing to an `assert`? That way, if the code changes (and someone adds an enum that doesn't adhere to the current style), the build will fail?

Fortunately, the current code has a very consistent style, which makes it very easy to add checks like these to avoid the "parsing" doing something unexpected.

Reply:
Yes, that would be good. Maybe also adding an abort message to notify the user what's wrong.

Reply:
Do you want an `if` then + a raise of a specific exception then (vs. `AssertionError`)?

Reply:
if + raise (or print + exit(1)) sounds good.

Reply:
Done!